File: GeneratedEntityReaderClasses/Components/PhysicsPickUpComponent.cs
namespace GeneratedNoitaClasses;
public class PhysicsPickUpComponent : NoitaComponentBase,  iNoitaType<PhysicsPickUpComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaCXForm<NoitaFloat> transform { get; set; } = new();
	public NoitaCVector2<NoitaFloat> original_left_joint_pos { get; set; } = new();
	public NoitaCVector2<NoitaFloat> original_right_joint_pos { get; set; } = new();
	public NoitaFloat pick_up_strength { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		transform.Read(s);
		original_left_joint_pos.Read(s);
		original_right_joint_pos.Read(s);
		pick_up_strength.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		transform.Write(s);
		original_left_joint_pos.Write(s);
		original_right_joint_pos.Write(s);
		pick_up_strength.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/PhysicsPickUpComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaCXForm
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class PhysicsPickUpComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    transform:NoitaCXForm = field(default_factory=lambda: NoitaCXForm(NoitaFloat))
    original_left_joint_pos:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    original_right_joint_pos:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    pick_up_strength:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.transform.Read(s)
        self.original_left_joint_pos.Read(s)
        self.original_right_joint_pos.Read(s)
        self.pick_up_strength.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.transform.Write(s)
        self.original_left_joint_pos.Write(s)
        self.original_right_joint_pos.Write(s)
        self.pick_up_strength.Write(s)

